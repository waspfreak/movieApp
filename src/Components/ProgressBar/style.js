import styled from 'styled-components';

export const Container = styled.div`
	background: #111;
	border-radius: 50%;
	padding: 2px;
`;

export const Svg = styled.svg`
	display: block;
	margin: 0px auto;
	max-width: 100%;
`;

export const CircleBg = styled.circle`
	fill: none;
`;

export const Circle = styled.circle`
	fill: none;
`;

export const CircleText = styled.text`
	font-size: 14px;
	text-anchor: middle;
	fill: white;
	font-weight: bold;
	color: white;
`;
