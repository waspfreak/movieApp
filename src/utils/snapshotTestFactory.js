import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
