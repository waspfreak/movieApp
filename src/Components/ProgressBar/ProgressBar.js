import React, {useEffect, useState, useRef} from 'react';
import PropTypes from 'prop-types';
import {Svg, CircleBg, Circle, CircleText, Container} from './style';

/**
 * Progress Bar UI component for user interaction
 */

export const ProgressBar = (props) => {
	const [offset, setOffset] = useState(0);
	const circleRef = useRef(null);
	const {size, progress, strokeWidth, circleOneStroke, circleTwoStroke} = props;

	const center = size / 2;
	const radius = size / 2 - strokeWidth / 2;
	const circumference = 2 * Math.PI * radius;

	useEffect(() => {
		const progressOffset = ((100 - progress) / 100) * circumference;
		setOffset(progressOffset);

		circleRef.current.style = 'transition: stroke-dashoffset 850ms ease-in-out';
	}, [setOffset, progress, circumference, offset]);

	return (
		<Container>
			<Svg width={size} height={size}>
				<CircleBg
					stroke={circleOneStroke}
					cx={center}
					cy={center}
					r={radius}
					strokeWidth={strokeWidth}
				/>
				<Circle
					ref={circleRef}
					stroke={circleTwoStroke}
					cx={center}
					cy={center}
					r={radius}
					strokeWidth={strokeWidth}
					strokeDasharray={circumference}
					strokeDashoffset={offset}
				/>
				<CircleText x={`${center}`} y={`${center}`}>
					{progress}%
				</CircleText>
			</Svg>
		</Container>
	);
};

ProgressBar.propTypes = {
	size: PropTypes.number.isRequired,
	progress: PropTypes.string.isRequired,
	strokeWidth: PropTypes.number.isRequired,
	circleOneStroke: PropTypes.string.isRequired,
	circleTwoStroke: PropTypes.string.isRequired,
};

export default ProgressBar;
