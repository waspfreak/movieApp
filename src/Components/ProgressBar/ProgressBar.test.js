import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {ProgressBar} from './ProgressBar';
const props: ProgressBar[] = [
	{
		size: 80,
		progress: '55',
		strokeWidth: 4,
		circleOneStroke: '#3c0a0adb',
		circleTwoStroke: '#111',
	},
];
snapshotTestFactory(ProgressBar, props);
