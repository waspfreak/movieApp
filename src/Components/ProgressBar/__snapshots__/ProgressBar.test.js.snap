// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should match snapshot 1`] = `
.c0 {
  background: #111;
  border-radius: 50%;
  padding: 2px;
  display: inline-block;
}

.c1 {
  display: block;
  margin: 0px auto;
  max-width: 100%;
}

.c2 {
  fill: none;
}

.c3 {
  fill: none;
}

.c4 {
  font-family: 'Titillium Web',sans-serif;
  font-size: 14px;
  text-anchor: middle;
  fill: white;
  font-weight: bold;
  color: white;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1"
      height="80"
      width="80"
    >
      <circle
        class="c2"
        cx="40"
        cy="40"
        r="38"
        stroke="#3c0a0adb"
        stroke-width="4"
      />
      <circle
        class="c3"
        cx="40"
        cy="40"
        r="38"
        stroke="#111"
        stroke-dasharray="238.76104167282426"
        stroke-dashoffset="107.44246875277092"
        stroke-width="4"
        style="transition: stroke-dashoffset 850ms ease-in-out;"
      />
      <text
        class="c4"
        x="40"
        y="40"
      >
        55
        %
      </text>
    </svg>
  </div>
</div>
`;
